src/analysis/normed_space/basic.lean

```diff
@@ -1947,6 +1947,98 @@ summable_of_nnnorm_bounded _ hf (assume i, le_refl _)
 
 end summable
 
+section cauchy_product
+
+/-! ## Multipliying two infinite sums in a normed ring
+
+In this section, we prove various results about `(∑' x : ι, f x) * (∑' y : ι', g y)` in a normed
+ring. There are similar results proven in `topology/algebra/infinite_sum` (e.g `tsum_mul_tsum`),
+but in a normed ring we get summability results which aren't true in general.
+
+We first establish results about arbitrary index types, `β` and `γ`, and then we specialize to
+`β = γ = ℕ` to prove the Cauchy product formula (see `tsum_mul_tsum_nat_of_summable_norm`).
+
+### Arbitrary index types
+-/
+
+variables {ι' : Type*} [normed_ring α]
+
+open_locale classical
+
+lemma real.summable_mul_of_summable_of_nonneg {f : ι → ℝ} {g : ι' → ℝ}
```

Comment:
At a glance it's not obvious to me why this proof requires f to index the reals; does it generalize?

Reply:
The main issue is that, to avoid order diamonds, I think (tell me if I'm wrong) I would need to define a class `conditionally_complete_linear_ordered_ring` which sounds a bit overspecific

Reply:
Sounds almost like #3292 ;). The answer to my question seems to simply be "because `summable_of_sum_le` only applies to the reals". I think @adomani's order refactor might solve that, but for now what you have here is the only sensible choice.